redash/handlers/base.py

```diff
@@ -66,10 +66,11 @@ def require_fields(req, fields):
 
 
 def get_object_or_404(fn, *args, **kwargs):
-    try:
-        return fn(*args, **kwargs)
-    except DoesNotExist:
+    rv = fn(*args, **kwargs)
+    if rv is None:
```

Comment:
its `.get()` and `.first()` methods return None when there's no entry, so I believe that's correct.